superdesk/core/privileges.py

```diff
@@ -0,0 +1,45 @@
+from dataclasses import dataclass
+from quart_babel.speaklater import LazyString
+
+
+@dataclass(frozen=True)
+class Privilege:
+    name: str
+    label: LazyString | None = None
+    category: LazyString | None = None
+    description: LazyString | None = None
+
+
+class PrivilegesRegistry:
+    __privileges: set[Privilege] | frozenset[Privilege]
+
+    def __init__(self):
+        self.__privileges = set()
+
+    def add(self, privilege: Privilege):
+        """Add a privilege if the registry is not locked."""
+
+        if self.is_locked:
+            raise RuntimeError("Cannot add privileges after the app has started.")
+
+        self.__privileges.add(privilege)  # type: ignore[union-attr]
```

Comment:
I decided to ignore the mypy warning as this approach with the property `is_locked` feels more pythonic and natural to me :)